src/Admin/AdminInterface.php

```diff
@@ -204,192 +165,136 @@ public function hasParentFieldDescription();
      *
      * @deprecated since sonata-project/admin-bundle 3.9, to be removed in 4.0
      */
-    public function trans($id, array $parameters = [], $domain = null, $locale = null);
+    public function trans($id, array $parameters = [], $domain = null, $locale = null): string;
 
     /**
      * Returns the parameter representing request id, ie: id or childId.
-     *
-     * @return string
      */
-    public function getIdParameter();
+    public function getIdParameter(): string;
 
     /**
      * Returns true if the route $name is available.
-     *
-     * @param string $name
-     *
-     * @return bool
      */
-    public function hasRoute($name);
+    public function hasRoute(string $name): bool;
 
-    public function setSecurityHandler(SecurityHandlerInterface $securityHandler);
+    public function setSecurityHandler(SecurityHandlerInterface $securityHandler): void;
 
-    /**
-     * @return SecurityHandlerInterface|null
-     */
-    public function getSecurityHandler();
+    public function getSecurityHandler(): ?SecurityHandlerInterface;
 
     /**
-     * @param string      $name
-     * @param object|null $object
-     *
-     * @return bool
+     * @param string $name
      */
-    public function isGranted($name, $object = null);
+    public function isGranted($name, ?object $object = null): bool;
```

Comment:
```suggestion
    public function isGranted(string $name, ?object $object = null): bool;
```

Reply:
See https://github.com/sonata-project/SonataAdminBundle/pull/6005#discussion_r419113584.

Reply:
Ok !
The phpDoc of the interface was saying `@param string $name` ; can we update it to `@param string|array $name` in order to avoid the mistake ?

Reply:
These kind of controversial decisions is what I want to follow in other PR, since it depends of what we want about this method. Maybe, regardless the types supported in the backend method, we want to narrow the type to be only `string` here.

Reply:
I understand your point.

But, If I was the one who made the PR, I would have totally missed the point that `isGranted` can work with array ; and since the Interface have only `@param string $name` as type, I would have updated the function type.
I almost sure my PR would have been accepted by someone who could have miss the fact `isGranted` works with array too.

Updating the phpDoc is one solution to avoid this kind of mistake.

Or maybe you prefer to add something like a `NEXT_MAJOR: Decide if we support string or string|array` ? In the same way we could add `NEXT_MAJOR: Decide if we support object or ?object` for the method of `AbstractAdmin`.

I agree there is *controversial decision*. What is the best way to be sure to take a decision about them ?